Update finder signup content item presenter

Seraphiana opened the pull request:
- Deprecated 'format' and replaced it with 'schema_name' and 'document_type'
- Format 'public_updated_at' to rfc3339 string, as this is the default
  requirement of content-schema validation.

[Trello Card](https://trello.com/c/HnpJ2PQf/147-create-missing-schema-for-finder-email-signup-medium),

[Depends On](https://github.com/alphagov/govuk-content-schemas/pull/327)

Rosa-Fox committed:
Update finder signup content item presenter

- Deprecated 'format' and replaced it with 'schema_name' and 'document_type'
- Format 'public_updated_at' to rfc3339 string, as this is the default
requirement of content-schema validation.

tuzz commented on spec/presenters/finders/finder_signup_content_item_presenter_spec.rb:
```diff
@@ -0,0 +1,16 @@
+      expect(presented_data).to be_valid_against_schema("finder_email_signup")
+    end
+  end
+end
```
I was thinking it might be nice to iterate through all the schemas in `lib/documents/schemas` and validate them in this test. I think you could do `Dir["lib/documents/schemas/*.json"].each` or something similar.

I think this would help to future-proof us if we decide to make changes to the content schema and want to check that all of our schemas still validate. What do you think?

Rosa-Fox committed:
Improved wording in logger

- Index the correct key to display names of finders that are not being published

tuzz committed:
Make presenter reload correctly when in isolation
